### DISCUSSION TITLE
Rendering issue on Windows 11

### DISCUSSION THREAD

**Discussion by cheddarking** [TomSchimansky/CustomTkinter#1720] [Announcements]
I have written a program for my windows users and it works fine on Windows 10, however on some (not all) Windows 11 users' machines some rendering issues occur. Please see below

**Working on Windows 11**

![image](https://github.com/TomSchimansky/CustomTkinter/assets/31318846/406e2ec9-a7b7-40ea-bb52-7c6133eff1e7)

**Not working on Windows 11**

![image](https://github.com/TomSchimansky/CustomTkinter/assets/31318846/40787370-fd3b-44b7-937f-60b253170072)

As you can see some of the text and hyperlinks get garbled. What could be causing this?

**Comment by octimot** [comment]
What fonts are you using in your theme?

**Comment by cheddarking** [comment]
> What fonts are you using in your theme?

The copyright label uses `ctk.CTkFont(family="Inter", weight="normal" )`

The User Manual and csv template links both use `ctk.CTkFont(family="Inter", size=12, weight="normal" )`

**Comment by ghost** [comment]
@cheddarking - It looks like the fonts are the problem. I think we can rule out OS or OS version as you mentioned that some Windows 11 users _are_ able to run the program properly.

Maybe the users (those facing issues) don't have the font installed. How are you providing the font? Is it being loaded or are you assuming the font is installed?

I also don't think this is a CustomTkinter issue. A better place for this would've been in the Discussions tab of this repository. Perhaps the Q&A section.

**Comment by ghost** [comment]
@cheddarking - Just checked [this page](https://learn.microsoft.com/en-us/typography/fonts/windows_11_font_list), the font Inter is _not_ installed on Windows 11 by default. The problem may be that you are assuming the font to be installed rather than providing it with the app.

Interesting, Windows 10 also does not have Inter installed... I guess my initial assumptions were incorrect. I am a macOS user, so I doubt I'll be of much help.

**Comment by TomSchimansky** [comment]
@cheddarking there is just a different font used than you intended, either you have to install it properly or the name is not specific enough.

**Comment by vednig** [comment]
Python usually loads via CPU and so does tkinter but loading graphics requires lot of computation plus single threaded programming and kernel crashes on Windows 11 can lead to performance issues.
For more help refer to [this](https://stackoverflow.com/questions/62608474/how-to-make-my-python-script-run-on-my-gpu)

**Reply by ghost** [reply]
I don't think there was any mention of performance issues or crashing...?

Check the OP again.

**Reply by vednig** [reply]
@cheddarking states "some(not all)" and most probably seems like sometimes it works and sometimes it doesn't which can only amount to internal windows performance and loading of font. When multiple fonts are loaded in tkinter, this issue seems to occur.
That's why I've suggested the above solution.

**Reply by vednig** [reply]
How do you think other users are having no issues? I mean how could they have a font installed in their system if it is not shipped with. Other than people especially installing it (which I don't think is the case)?

**Reply by ghost** [reply]
@vednig - Understood, thank you for explaining :)

**Reply by cheddarking** [reply]
To be clear the program does not crash it's just the font display is incoherent.

**Reply by cheddarking** [reply]
Is there a way to ship the fonts with the Pyinstaller compilation?

**Reply by ghost** [reply]
@cheddarking - using the `--add-data` option... Try reading the [Pyinstaller documentation](https://pyinstaller.org/en/stable/usage.html) for more information.

**Comment by Prakash4844** [comment]
Most likely, the font are just not installed on your machine, either install those fonts or use a different font altogether to test things out, This is not windows problem.

**Reply by ghost** [reply]
It was other users of the author's app that were facing the issue. And to install a font to use an application doesn't make sense. IMOP, a better solution would be to provide the font (bundled) with the application.

**Reply by Prakash4844** [reply]
> a better solution would be to provide the font (bundled) with the application.

Yes, you are right the fonts should be packaged along with the application. but even if the font was not found at runtime, it should fallback to using default system fonts. instead of just picking a random font(emojis?).

**Reply by cheddarking** [reply]
This error is from one of my end users. This works fine on my machine. Maybe something unique about their machine..

**Reply by ghost** [reply]
@cheddarking - A bundled application shouldn't rely on the user's environment... Except for things like installed fonts.

**Comment by ghost** [comment]
>  but even if the font was not found at runtime, it should fallback to using default system fonts. instead of just picking a random font(emojis?).

That makes sense... But there is no traceback, so no exception is being thrown... Meaning the font is being found?

**Reply by Prakash4844** [reply]
That is indeed strange behavior, maybe the program does throw an exception but it is not shown in foreground. might be a good idea to execute it through terminal. to see if anything is comes up.

also, i found that it might be useful to just add the font to application using pyglet.

You can try `pyglet.font.add_file()` from pyglet module. But make sure you have the right to distribute the font.

source: [stackoverflow](https://stackoverflow.com/questions/68750075/python-tkinter-how-to-use-a-font-which-is-not-installed-on-a-different-pc-whil)

**Reply by ghost** [reply]
I think instead of introducing a whole new module (`pyglet`) just to load a font, we can load a font with the following line:
```python
customtkinter.FontManager.load_font(...)
```

**Reply by ghost** [reply]
> That is indeed strange behavior, maybe the program does throw an exception but it is not shown in foreground. might be a good idea to execute it through terminal. to see if anything is comes up.

Unless the author (of this issue) disabled the traceback when building with Pyinstaller (it's possible with an option), a traceback would've popped up in a new terminal window.

**Comment by cheddarking** [comment]
Thanks for all your contributions, sorry I have been out of the loop. I just specified this font size assuming that it was supported by CustomTkinter. Interestingly enough it works on both Windows 10 and 11 but not some 11 users. Not sure why that is. Is there a list of fonts that are safe/supported in CustomTkinter?

**Reply by ghost** [reply]
Maybe Roboto. As, looking at `customtkinter`'s source, Roboto is a `.ttf` file that is provided.

**Reply by cheddarking** [reply]
Thanks I think  I will default to that then. I use it on the other elements that are working for the user.
